server/src/tree-sitter/analyzer.ts

```diff
@@ -274,6 +275,50 @@ export default class Analyzer {
     (n?.type === ':' && n?.parent?.type === 'ERROR' && n?.parent?.previousSibling?.type === 'override') // when having MYVAR:append: = '123' and the second or later colon is typed
   }
 
+  public isInsidePythonRegion (
+    uri: string,
+    line: number,
+    column: number
+  ): boolean {
+    let n = this.nodeAtPoint(uri, line, column)
+
+    while (n?.parent !== null && n?.parent !== undefined) {
+      if (TreeSitterUtils.isInlinePython(n.parent) || TreeSitterUtils.isPythonDefinition(n.parent)) {
+        return true
+      }
+      n = n.parent
+    }
+
+    return false
+  }
+
+  public isPythonDatastoreVariable (
+    uri: string,
+    line: number,
+    column: number
+  ): boolean {
+    const n = this.nodeAtPoint(uri, line, column)
+    if (!this.isInsidePythonRegion(uri, line, column)) {
+      return false
+    }
+    // Example:
+    // n.text: FOO
+    // n.parent.text: 'FOO'
+    // n.parent.parent.text: ('FOO')
+    // n.parent.parent.parent.text: d.getVar('FOO')
+    const parentParentParent = n?.parent?.parent?.parent
+    if (parentParentParent?.type !== 'call') {
+      return false
+    }
+    const match = parentParentParent.text.match(/^(d|e\.data)\.(?<name>.*)\((?<params>.*)\)$/) // d.name(params), e.data.name(params)
+    const functionName = match?.groups?.name
+    if (functionName === undefined || !(bitBakeDocScanner.pythonDatastoreFunction.includes(functionName))) {
+      return false
+    }
+    const variable = match?.groups?.params?.split(',')[0]?.trim().replace(/('|")/g, '')
+    return variable === n?.text
```

Comment:
do these functions ever take more than one param? If they do, the params after the first one will never pass the check unless they are the same as the first one. If they don't, then seems like `variable` is always the same as `n.text` at this point.

Reply:
These functions can have more than one parameter, but usually only the first parameter can be a "yocto/bitbake variable".`

For example, `d.setVar("BZRDIR", "CVSDIR")` will set `BZRDIR` to the string `"CVSDIR"`. If you actually want to set it to the value of `CVSDIR`, you'd have to do `d.setVar("BZRDIR", d.getVar("CVSDIR"))`.

One exception to this might be `d.renameVar("BZRDIR", "CVSDIR")`. Maybe I should treat it as a special case.

Reply:
I'll merge as it is for now and add a note for renameVar in my ticket so it has to be handled before it can be said to be done